src/main/java/edu/greenblitz/tobyDetermined/commands/NodeToNeighbourCommand.java

```diff
@@ -0,0 +1,102 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class NodeToNeighbourCommand extends GBCommand {
+    private NodeBase nodeBase;
+    private final Extender extender;
+    private final Elbow elbowSub;
+    private NodeArm start;
+    private NodeArm end;
+    private static final double COMBINED_VELOCITY = 2.3;
+    private static final double MAX_EXTENDER_VELOCITY = 1; //In Meters Per Second
+    private static final double MAX_ANGULAR_VELOCITY = 1.5;//In Radians Per Second
+
+    public NodeToNeighbourCommand(NodeArm start, NodeArm end){
+        extender = Extender.getInstance();
+        elbowSub = Elbow.getInstance();
+        nodeBase = NodeBase.getInstance();
+        //require(elbowSub);
+        //require(extender);
+        this.start = start;
+        this.end = end;
+    }
+
+
+    public static double cosineRule(double sideA, double sideB, double angleBetweenSideAndSideB) {
+        return Math.sqrt(sideA*sideA+sideB*sideB-2*sideA*sideB*Math.cos(angleBetweenSideAndSideB));
+    }
+    public static double getRatioBetweenAngleAndLength(double a, double b, double gamma) {
+        if (b == 0)
+            b+=0.02;//weird edge case where 0 * big number = 0 so messes up whole calculations
+        double c = cosineRule(a,b,gamma);
+        double height = b*Math.sin(gamma);
+        double adjacent = Math.sqrt(c*c - height*height);
+        return height/adjacent;
+    }
+
+    public double calculateExtenderVelocity(double ratio, NodeArm nodeEndIndex) {
+        double signOfExtender = Math.signum(nodeEndIndex.getExtendPos()-extender.getLength());
+        double extenderVelocity = Math.sqrt(COMBINED_VELOCITY * COMBINED_VELOCITY /(ratio*ratio+1));
+        return signOfExtender * extenderVelocity;
+    }
+    public double calculateAngularVelocity(double startVelocity,NodeArm nodeEndIndex){
+
+        double signOfAngle = Math.signum(nodeEndIndex.getAnglePos()-elbowSub.getAngleRadians());
+        double magnitudeOfVelocity = startVelocity/extender.getLength();
+        magnitudeOfVelocity = Math.min(MAX_ANGULAR_VELOCITY,magnitudeOfVelocity);
+        magnitudeOfVelocity = Math.max(-MAX_ANGULAR_VELOCITY,magnitudeOfVelocity);
+        return signOfAngle*Math.abs(magnitudeOfVelocity);
+    }
+    public void moveArm( NodeArm nodeEndIndex){
+        double start = extender.getLength();
+        double end = nodeEndIndex.getExtendPos();
+        double gamma = nodeEndIndex.getAnglePos()-elbowSub.getAngleRadians();
+        double ratio = getRatioBetweenAngleAndLength(start,end,gamma);
+        double extenderVelocity = calculateExtenderVelocity(ratio,nodeEndIndex);
+        double angularVelocity = calculateAngularVelocity(ratio*extenderVelocity,nodeEndIndex);
+        extenderVelocity = Math.min(MAX_EXTENDER_VELOCITY,extenderVelocity);
+        extenderVelocity = Math.max(-MAX_EXTENDER_VELOCITY,extenderVelocity);
+        if (!(NodeBase.getInstance().getIfInAngle(elbowSub.getAngleRadians(),nodeEndIndex))){
```

Comment:
why does this if else exist for the elbow but not for extender

Reply:
the length can be in right length change and then return to right length in purpose to make straight line but angle not